### PR TITLE
Add support for backend buckets, cloud functions

### PR DESCRIPTION
**Pull request by slevenick**
Add singular and plural resources as well as docs and tests for compute backend buckets and cloud functions.

Addresses https://github.com/inspec/inspec-gcp/issues/124 and https://github.com/inspec/inspec-gcp/issues/121

### CODE REVIEW

File: test/integration/verify/controls/google_cloudfunctions_cloud_functions.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#     CONTRIBUTING.md located at the root of this package.
+#
+# ----------------------------------------------------------------------------
+
+title 'Test GCP google_cloudfunctions_cloud_functions resource.'
+
+gcp_project_id = attribute(:gcp_project_id, default: 'gcp_project_id', description: 'The GCP project identifier.')
+cloudfunction = attribute('cloudfunction', default: {
+  "name": "inspec-gcp-function",
+  "location": "europe-west1",
```
Comment (skpaterson) [inline test/integration/verify/controls/google_cloudfunctions_cloud_functions.rb]: Same as above, would be great to use the global location variable here.

File: test/integration/verify/controls/google_cloudfunctions_cloud_function.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+#     CONTRIBUTING.md located at the root of this package.
+#
+# ----------------------------------------------------------------------------
+
+title 'Test GCP google_cloudfunctions_cloud_function resource.'
+
+gcp_project_id = attribute(:gcp_project_id, default: 'gcp_project_id', description: 'The GCP project identifier.')
+cloudfunction = attribute('cloudfunction', default: {
+  "name": "inspec-gcp-function",
+  "location": "europe-west1",
```
Comment (skpaterson) [inline test/integration/verify/controls/google_cloudfunctions_cloud_function.rb]: Can we use the global location variable here for consistency with a single chosen region?

Comment (slevenick) [reply test/integration/verify/controls/google_cloudfunctions_cloud_function.rb]: Ah yeah, so cloud functions only has a certain set of locations that it is available in. See: https://cloud.google.com/functions/docs/locations

So I figure it should be configured separately

**Comment by ali-himindz** [comment]
Great work @slevenick. Are you pushing the changes recommended by @skpaterson  soon? 
Hope we get this PR merged ASAP.
thanks
Ali

File: test/integration/verify/controls/google_cloudfunctions_cloud_function.rb (hunk 2 above)
Comment (skpaterson) [reply test/integration/verify/controls/google_cloudfunctions_cloud_function.rb]: Sure, that makes sense.  Can we keep this as a non-MM yaml variable for convenience?  Would be good to update the README e.g. adding the new variable such as `GCP_CLOUD_FUNCTION_REGION` to the instructions on changing region.

Comment (slevenick) [reply test/integration/verify/controls/google_cloudfunctions_cloud_function.rb]: Added to the config file and README

Comment (skpaterson) [reply test/integration/verify/controls/google_cloudfunctions_cloud_function.rb]: Fantastic, thanks @slevenick !

**Comment by ali-himindz** [comment]
thanks for @skpaterson  for quick action. Much appreciated.

File: test/integration/verify/controls/google_cloudfunctions_cloud_function.rb (hunk 2 above)
Comment (ali-himindz) [reply test/integration/verify/controls/google_cloudfunctions_cloud_function.rb]: Thanks @slevenick.

**Pull request merged**